File: mycollab-web/src/main/java/com/esofthead/mycollab/module/user/accountsettings/view/parameters/UserScreenData.java
package com.esofthead.mycollab.module.user.accountsettings.view.parameters;

import com.esofthead.mycollab.module.user.domain.SimpleUser;
import com.esofthead.mycollab.module.user.domain.User;
import com.esofthead.mycollab.module.user.domain.criteria.UserSearchCriteria;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

public class UserScreenData {
	public static class Read extends ScreenData<Object> {

		public Read(Object params) {
			super(params);
		}
	}

	public static class Add extends ScreenData<SimpleUser> {

		public Add(SimpleUser params) {
			super(params);
		}
	}

	public static class Edit extends ScreenData<User> {

		public Edit(User params) {
			super(params);
		}
	}

	public static class Search extends ScreenData<UserSearchCriteria> {

		public Search(UserSearchCriteria params) {
			super(params);
		}
	}
}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/user/accountsettings/view/parameters/UserScreenData.scala
package com.esofthead.mycollab.module.user.accountsettings.view.parameters

import com.esofthead.mycollab.module.user.domain.{User, SimpleUser}
import com.esofthead.mycollab.module.user.domain.criteria.UserSearchCriteria
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd.
 * @since 5.1.0
 */
object UserScreenData {

  class Read(params:String) extends ScreenData[String](params) {}

  class Add(params:SimpleUser) extends ScreenData[SimpleUser](params) {}

  class Edit(params:User) extends ScreenData[User](params) {}

  class Search(params:UserSearchCriteria) extends ScreenData[UserSearchCriteria](params) {}
}
